ebean-core/src/main/java/io/ebeaninternal/server/executor/DefaultBackgroundExecutor.java

```diff
@@ -1,119 +1,147 @@
+  public DefaultBackgroundExecutor(int schedulePoolSize, int shutdownWaitSeconds, String namePrefix, BackgroundExecutorWrapper wrapper) {
     this.schedulePool = new DaemonScheduleThreadPool(schedulePoolSize, shutdownWaitSeconds, namePrefix + "-periodic-");
     this.pool = new DaemonExecutorService(shutdownWaitSeconds, namePrefix);
+    this.wrapper = wrapper;
+    logger.debug("Created backgroundExecutor {} (schedulePoolSize={}, shutdownWaitSeconds={})", namePrefix, schedulePoolSize, shutdownWaitSeconds);
   }
 
   /**
    * Wrap the task with MDC context if defined.
    */
-  <T> Callable<T> wrapMDC(Callable<T> task) {
-    final Map<String, String> map = MDC.getCopyOfContextMap();
-    if (map == null) {
+  <T> Callable<T> wrap(Callable<T> task) {
+    if (wrapper == null) {
       return task;
     } else {
-      return () -> {
-        MDC.setContextMap(map);
-        try {
-          return task.call();
-        } finally {
-          MDC.clear();
-        }
-      };
+      return wrapper.wrap(task);
     }
   }
 
   /**
    * Wrap the task with MDC context if defined.
    */
-  Runnable wrapMDC(Runnable task) {
-    final Map<String, String> map = MDC.getCopyOfContextMap();
-    if (map == null) {
+  Runnable wrap(Runnable task) {
+    if (wrapper == null) {
       return task;
     } else {
-      return () -> {
-        MDC.setContextMap(map);
-        try {
+      return wrapper.wrap(task);
+    }
+  }
+  
+
+  /**
+   * Decorates a runnable by adding an exception handler and some timing metrics.
+   * This is used in methods that accepts a <code>Runnable</code> and return
+   * either <code>void</code> or <code>ScheduledFuture</code>, as there is
+   * normally no Future.get() call.
+   * 
+   * Note: When submitting a <code>Callable</code>, you must check
+   * <code>Future.get()</code> for exceptions.
+   */
+  private Runnable logExceptions(Runnable task) {
+    long queued = System.nanoTime();
+    logger.trace("Queued {}", task);
+    return () -> {
+      try {
+        if (logger.isTraceEnabled()) {
+          long start = System.nanoTime();
+          logger.trace("Start {} (delay time {} us)", task, (start - queued) / 1000L);
+          task.run();
+          logger.trace("Stop {} (exec time {} us)", task, (System.nanoTime() - start) / 1000L);
+        } else {
           task.run();
-        } finally {
-          MDC.clear();
         }
-      };
-    }
+      } catch (Throwable t) {
+        // log any exception here. Note they will not bubble up to the calling user
+        // unless Future.get() is checked. (Which is almost never done on scheduled
+        // background executions)
+        logger.error("Error while executing the task {}", task, t);
+        throw t;
+      }
+    };
   }
 
   @Override
   public <T> Future<T> submit(Callable<T> task) {
-    return pool.submit(wrapMDC(task));
+    return pool.submit(wrap(task));
```

Comment:
Maybe add `// Note: Here is no "logExceptions", because it is intended to check Future.get() by the invoker`